src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -392,30 +393,44 @@ private SnapshotProject retrieveSnapshotProject(String sql, MapSqlParameterSourc
         String sort,
         String direction,
         String filter,
+        List<UUID> datasetIds,
         List<UUID> accessibleSnapshotIds) {
         logger.debug("retrieve snapshots offset: " + offset + " limit: " + limit + " sort: " + sort +
-            " direction: " + direction + " filter:" + filter);
+            " direction: " + direction + " filter: " + filter + " datasetIds: " + StringUtils.join(datasetIds, ","));
         MapSqlParameterSource params = new MapSqlParameterSource();
         List<String> whereClauses = new ArrayList<>();
-        DaoUtils.addAuthzIdsClause(accessibleSnapshotIds, params, whereClauses);
+        DaoUtils.addAuthzSnapshotIdsClause(accessibleSnapshotIds, params, whereClauses);
         whereClauses.add(" flightid IS NULL"); // exclude snapshots that are exclusively locked
 
         // add the filter to the clause to get the actual items
         DaoUtils.addFilterClause(filter, params, whereClauses);
-        String whereSql = "";
-        if (!whereClauses.isEmpty()) {
-            whereSql = " WHERE " + StringUtils.join(whereClauses, " AND ");
+        String joinSql = "";
+
+        if (!datasetIds.isEmpty()) {
+            joinSql = " JOIN snapshot_source ON snapshot.id = snapshot_source.snapshot_id ";
+            String datasetMatchSql =
+                datasetIds.stream()
+                    .map(datasetId -> " snapshot_source.dataset_id = '" + datasetId + "'")
```

Comment:
Wouldn't you want to write this as an `in`?  Also this is a sql injection risk I think.  You never want to reference values that a user can pass in like this

Reply:
`IN` may be a better way to express this. Regarding SQL injection there's no real risk unless there's a way to make a `UUID` that breaks SQL -- the `String` api input is converted to UUID above. It's true that building a query like this is usually not good, and changing it to `IN` would let you use JDBC templating.

Reply:
Ah, fair point about the UUID format...I just get nervous with appending values in raw in the sql

Reply:
does this make sense to do as a follow on? It seems like there are additional pieces I'd like to fold iin